### PR TITLE
feat:create service for validation

### PR DESCRIPTION
**Pull request by LeVHoangduc**
@lenguyenthimy-agilityio @hungnguyenhuu-agilityio Please help me to check and merge it

### CODE REVIEW

File: src/javascripts/models/users.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import LoginService from "../services/loginService";
+import User from "./user";
+
+class Users {
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/users.js]: This one is useless. Please help to recheck it

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated it. Please help me to check again.

File: src/javascripts/services/loginService.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+   * Get user list from database and returns for users model.
+   * @returns {Object} User object
+   */
+  getUserList = async () => {
+    const data = await this.apiRequest.get();
+    return data;
+  };
+}
+
+export default LoginService;
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/loginService.js]: As we discussed before, we don't define the login service, right? So why we have this one

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated my service. Please help me to check again.

File: src/javascripts/app.js [hunk 3]
```diff
@@ -1 +1,18 @@
-// TODO: implement App Object
+import View from "./views/view";
+import Model from "./models/model";
+import Controller from "./controllers/controller";
+
+export class App {
+  /**
+   * Constructor off App object
+   */
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/app.js]: This block of comment is useless. Pls remove it

File: src/javascripts/services/apiService.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    });
+    if (response.ok) {
+      return await response.json();
+    } else {
+      throw new Error("Error while sending request");
+    }
+  };
+}
+
+export default ApiService;
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/apiService.js]: This is the branch for service validation but it include api service. Please recheck again

File: src/javascripts/services/validationService.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        errorEl.textContent = field.invalidMessage;
+        errorEl.classList.add("active");
+        isValid = false;
+      }
+
+      // If the field is valid, remove any error styling and message
+      else {
+        inputEl.classList.remove("error");
+        errorEl.textContent = "";
+        errorEl.classList.remove("active");
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]:
```
        inputEl.classList.remove("error");
        errorEl.textContent = "";
        errorEl.classList.remove("active");
        isValid = false;
```
This code block is duplicated. Did you see that?

File: src/javascripts/services/validationService.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      const loginForm = document?.querySelector(".login-form");
+      const emailInput = loginForm.email;
+      const emailError = emailInput.nextElementSibling;
+      const passwordInput = loginForm.password;
+      const passwordError = passwordInput.nextElementSibling;
+
+      // Object to store field validation data
+      const loginFields = [
+        { name: "email", regex: REGEX.EMAIL, error: emailError, requiredMessage: MESSAGE.EMAIL_REQUIRED, invalidMessage: MESSAGE.INVALID_EMAIL },
+        { name: "password", regex: REGEX.PASSWORD, error: passwordError, requiredMessage: MESSAGE.PASSWORD_REQUIRED, invalidMessage: MESSAGE.INVALID_PASSWORD },
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: Code style

File: src/javascripts/services/validationService.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+      const descriptionPhotoError = descriptionPhotoInput.nextElementSibling;
+
+      // Object to store field validation data
+      const cardFields = [
+        { name: "word", regex: REGEX.CONTENT, error: wordError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "type", regex: REGEX.CONTENT, error: typeError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "meaning", regex: REGEX.CONTENT, error: meaningError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "typeMeaning", regex: REGEX.CONTENT, error: typeMeaningError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "description", regex: REGEX.CONTENT, error: descriptionError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "descriptionPhoto", regex: REGEX.IMAGE, error: descriptionPhotoError, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_IMAGE },
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: Code style

File: src/javascripts/services/validationService.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
+import { MESSAGE, REGEX } from "../constants/constants";
+
+class ValidationService {
+  constructor() {}
+
+  /**
+   * Method to validate form data
+   * @param {String} validation
+   * @param {Object} object
+   * @returns
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: return?

File: src/javascripts/services/validationService.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      const typeInput = modalForm.type;
+      const typeError = typeInput.nextElementSibling;
+      const meaningInput = modalForm.meaning;
+      const meaningError = meaningInput.nextElementSibling;
+      const typeMeaningInput = modalForm.typeMeaning;
+      const typeMeaningError = typeMeaningInput.nextElementSibling;
+      const descriptionInput = modalForm.description;
+      const descriptionError = descriptionInput.nextElementSibling;
+      const descriptionPhotoInput = modalForm.descriptionPhoto;
+      const descriptionPhotoError = descriptionPhotoInput.nextElementSibling;
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: too much of constant definition. Look so complex. Pleas refactor them

Comment (LeVHoangduc) [reply src/javascripts/services/validationService.js]: @hungnguyenhuu-agilityio I have updated it.

File: src/javascripts/services/validationService.js (hunk 8 above)
Comment (LeVHoangduc) [reply src/javascripts/services/validationService.js]: @hungnguyenhuu-agilityio  I have updated it.

File: src/javascripts/services/validationService.js (hunk 7 above)
Comment (LeVHoangduc) [reply src/javascripts/services/validationService.js]: @hungnguyenhuu-agilityio In my editor ( VS code), It doesn't show such style. Should you help me how to fix it ?

File: src/javascripts/services/validationService.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+      const modalForm = document?.querySelector(".modal-add");
+
+      // Object to store field validation data
+      const cardFields = [
+        { name: "word", regex: REGEX.CONTENT, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "type", regex: REGEX.CONTENT, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "meaning", regex: REGEX.CONTENT, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "typeMeaning", regex: REGEX.CONTENT, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "description", regex: REGEX.CONTENT, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_CONTENT },
+        { name: "descriptionPhoto", regex: REGEX.IMAGE, requiredMessage: MESSAGE.CONTENT_REQUIRED, invalidMessage: MESSAGE.INVALID_IMAGE },
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: Code style

File: src/javascripts/services/validationService.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+  /**
+   * Method to validate form data
+   * @param {String} validation
+   * @param {Object} object
+   * @returns {Boolean} object is validated
+   */
+  formValidator = (validation) => (object) => {
+    if (validation === "card") {
+      const modalForm = document?.querySelector(".modal-add");
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: This is the service. Why we have the interaction with view here?

Comment (LeVHoangduc) [reply src/javascripts/services/validationService.js]: @hungnguyenhuu-agilityio I have updated it. Please help me to check again.

File: src/javascripts/services/validationService.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import { MESSAGE, REGEX } from "../constants/constants";
+import { helpers } from "../helpers/helpers";
+class ValidationService {
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js:3]: code style

File: src/javascripts/services/validationService.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+  constructor() {}
+
+  /**
+   * Method to validate form data
+   * @param {String} validation
+   * @param {HTMLElement} form
+   * @param {Object} object
+   * @returns {Boolean} object is validated
+   */
+  formValidator = (validation) => (form, object) => {
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/services/validationService.js]: The definition of this function is weird, why do you define like that?

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated. Please help me to check again

**Pull request merged**